### DISCUSSION TITLE
Using automatic sidebar menus with multiple top-level sections in `/content/en/`?

### DISCUSSION THREAD

**Discussion by johnnymatthews** [thuliteio/doks#817] [Q&A] [answered]
**TLDR: is there a way to automatically generate the sidebar menu for multiple topbar sections?**

I'm attempting to create two main sections (the top bar navigation) in a Doks site while using the automatic sidebar. Sorry, I don't have a public repo to show you yet.

Here's the structure I've got:

```plaintext
en
├── _index.md
├── alan
│   ├── _index.md
│   └── brian
│       ├── _index.md
│       ├── brian-commands.md
│       ├── brian-introduction.md
│       └── brian-quick-start.md
└── docs
    ├── _index.md
    ├── help
    │   ├── _index.md
    │   ├── faq.md
    │   ├── how-to-update.md
    │   └── troubleshooting.md
    └── prologue
        ├── _index.md
        ├── commands.md
        ├── introduction.md
        └── quick-start.md
```

Pages within the `alan` section contain the following `menu` front-matter:

```toml
menu:
  alan:
    parent: "brian"
```

My `menus.en.toml` file looks like this:

```toml
[[main]]
  name = "Docs"
  url = "/docs/prologue/introduction/"
  weight = 10

[[main]]
  name = "Alan"
  url = "/alan/brian/"
  weight = 10

[[docs]]
  name = "Prologue"
  weight = 10
  identifier = "prologue"
  url = "/docs/prologue/"

[[docs]]
  name = "Help"
  weight = 60
  identifier = "help"
  url = "/docs/help/"

[[alan]]
  name = "Brian"
  weight = 10
  identifier = "brian"
  url = "/alan/brian/"
```

Unfortunately, if I go to `/alan/brian/brian-introduction/`, I don't get a sidebar menu:

<img width="931" alt="image" src="https://user-images.githubusercontent.com/9611008/177399090-650a1602-241a-479c-9fad-792e616d25c9.png">

Is there a way to automate the sidebar menu for individual directories in `/content/en/`?

**Comment by h-enk** [comment, marked as answer]
Not sure what's wrong with your setup. It should work out of the box. If you have repo available, I could be of better help.

Can't replicate it. What I did on a fresh install:

- run `npm run create -- --kind docs tutorials`
- add to `./config/_default/menus/menus.en.toml`:
  ```toml
  [[main]]
    name = "Tutorials"
    url = "/tutorials/"
    weight = 15
  ```
- in `./config/_default/params.toml` set (needed for smaller screen sizes):
  ```toml
  [sections]
    sectionNav = ["docs", "tutorials"]
  ```
And, it "just" works: https://doks-817.netlify.app/tutorials/lorem/ipsum/

**Reply by johnnymatthews** [reply]
🤦 I wasn't using the `npm run create` commands. My bad. Works perfectly now! Thanks for the heads up!

![image](https://user-images.githubusercontent.com/9611008/177585304-dbd1e1a9-5c8d-46fa-87ea-a09f55300846.png)
